src/test/java/io/github/rabobank/shadow_tool/ShadowFlowTest.java

```diff
@@ -308,6 +313,46 @@ void shouldBeAbleToCompareCollectionOfObjects() {
         assertThatLogContains("The following differences were found: place, madrigals");
     }
 
+    @ParameterizedTest
+    @ValueSource(classes = {ArrayList.class, HashSet.class})
+    void typeOfCollectionListShouldBeTheResult(final Class<? extends Collection<?>> clazz) {
+        final var shadowFlow = createBlockingShadowFlow(100);
+
+        final var result = shadowFlow.compareCollections(
+                () -> createDummyCollectionOfType(clazz, dummyObjectA),
+                () -> createDummyCollectionOfType(clazz, dummyObjectB),
+                DummyObject.class
+        );
+
+        assertInstanceOf(clazz, result);
```

Comment:
I think I get what you mean now. This would work in the old situation too. For this to actually test the new flow, we would need to assign it to a specifically typed variable, instead of `var`. But I guess this is also fine